frame/asset-conversion/src/lib.rs

```diff
@@ -1141,6 +1141,45 @@ where
 
 		Ok(amount_in)
 	}
+
+	// If successful returns the amount out.
+	fn swap_exact_native_for_tokens(
+		sender: T::AccountId,
+		asset_id: T::AssetId,
+		amount_in: T::Balance,
+		amount_out_min: Option<T::AssetBalance>,
+		send_to: T::AccountId,
+		keep_alive: bool,
+	) -> Result<T::AssetBalance, DispatchError> {
+		ensure!(amount_in > Zero::zero(), Error::<T>::ZeroAmount);
+		if let Some(amount_out_min) = amount_out_min {
+			ensure!(amount_out_min > Zero::zero(), Error::<T>::ZeroAmount);
+		}
+		let mut path = sp_std::vec::Vec::new();
+		path.push(T::MultiAssetIdConverter::get_native());
+		path.push(T::MultiAssetIdConverter::into_multiasset_id(&asset_id));
```

Comment:
That's different, not related to XCM MultiAsset

Reply:
XCM is currently in polkadot repo. If it was moved to a separate repo then possibly it could be used directly by substrate pallets but for now this is not possible.